packages/common/src/services/dictionaryService.ts

```diff
@@ -0,0 +1,30 @@
+import { isEmpty } from 'lodash-es';
+import { Dependencies } from '../config/config.js';
+
+import getCategoryUtils from '../utils/categoryUtils.js';
+import getDictionaryUtils from '../utils/dictionaryUtils.js';
+
+const dictionaryService = (dependencies: Dependencies) => {
+	return {
+		register: async (categoryName: string, dictionaryName: string, version: string) => {
+			const { createCategoryIfDoesNotExist } = getCategoryUtils(dependencies);
+			const savedCategory = await createCategoryIfDoesNotExist(categoryName);
+			if (isEmpty(savedCategory)) return;
+
+			const { createDictionaryIfDoesNotExist, fetchDictionaryByVersion } = getDictionaryUtils(dependencies);
+			const dictionary = await fetchDictionaryByVersion(dictionaryName, version);
+			if (isEmpty(dictionary)) return;
+
+			const savedDictionary = await createDictionaryIfDoesNotExist(dictionaryName, version, savedCategory, dictionary);
+			return savedDictionary;
+		},
+
+		getCurrentDictionary: async () => {
+			// TODO
+			// Read directionary from DB
+			return '';
```

Comment:
Unimplemented, let's throw an error that makes this clear to make sure any uses of this do not appear to work but actually fail to do anything. Very helpful as the codebase grows.

Reply:
just in case, were talking "error 501"